app/models/package_manager/maven.rb

```diff
@@ -18,15 +19,41 @@ class Maven < Base
     }.freeze
 
     def self.package_link(project, version = nil)
+      if version
+        db_version = project.versions.find_by(number: version)
+        if db_version.repository_sources.include?(REPOSITORY_SOURCE_NAME)
+          return MavenUrl.from_name(project.name, repository_base).search(version)
+        else
+          repository_source = db_version.repository_sources.first
+          return "MavenManager::#{repository_source}".constantize.package_link(project, version)
```

Comment:
I can't help but be slightly nervous about this. Maybe we can add something like this to get em?
```
def get_source(source_name)
  if MavenManager.const_defined?(source_name)
    return MavenManager.const_get(source_name)
  end

  raise "Unknown source #{source_name}" 
end
```

Reply:
Though looking at that, it protects us from... nothing